modulemd/modulemd-defaults-v1.c

```diff
@@ -1213,60 +1215,69 @@ modulemd_defaults_v1_merge (const gchar *module_name,
   gpointer key, value;
   GHashTable *intent_profiles = NULL;
   GHashTable *merged_intent_profiles = NULL;
+  guint64 from_modified;
+  guint64 into_modified;
   g_autoptr (GHashTable) intent_profile_defaults = NULL;
   gchar *intent_name = NULL;
   gchar *intent_default_stream = NULL;
   gchar *merged_default_stream = NULL;
   g_autoptr (GError) nested_error = NULL;
+  const gchar *module_name =
+    modulemd_defaults_get_module_name (MODULEMD_DEFAULTS (into));
+
+  from_modified = modulemd_defaults_get_modified (MODULEMD_DEFAULTS (from));
+  into_modified = modulemd_defaults_get_modified (MODULEMD_DEFAULTS (into));
 
-  merged = modulemd_defaults_v1_new (module_name);
+  /* Start from a copy of "into" */
+  merged =
+    MODULEMD_DEFAULTS_V1 (modulemd_defaults_copy (MODULEMD_DEFAULTS (into)));
 
   /* Merge the default streams */
-  if (into->default_stream && !from->default_stream)
-    {
-      modulemd_defaults_v1_set_default_stream (
-        merged, into->default_stream, NULL);
-    }
-  else if (from->default_stream && !into->default_stream)
+  if (from->default_stream && !into->default_stream)
```

Comment:
```suggestion
  if (from->default_stream && !merged->default_stream)
```

Since you seemed to be deliberately changing the code to refer to `merged` (as a copy of `into`) throughout the block below, it should be changed here as well for consistency.